middlewares/authorizeRoles.js

```diff
@@ -0,0 +1,52 @@
+const {
+  ROLES: { SUPERUSER, APPOWNER, MEMBER },
+} = require("../constants/users");
+
+const VALID_ROLES = [SUPERUSER, APPOWNER, MEMBER];
+
+/**
+ * Check if the user has authorization based on their role.
+ * @param {Array.<String>} allowedRoles - Allowed roles for API consumption.
+ * @param {Object} userRoles - Roles information of the current user.
+ * @returns {Boolean} - Whether the current user is authorized or not.
+ */
+const userHasPermission = (allowedRoles, userRoles) => {
+  return allowedRoles.some((role) => userRoles[`${role}`] === true);
```

Comment:
1. Does this one-line function abstraction help us?
2. Isn't `userRoles[key]` already a 'true' boolean or undefined, `return allowedRoles.some((role) => userRoles[`${role}`]);` should work for us.